pkg/netutil/vpn.go

```diff
@@ -0,0 +1,130 @@
+package netutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type TailscaleOutput struct {
+	TailscaleIPs []string `json:"TailscaleIPs"`
+}
+
+// VPNInfo includes node information of the VPN. It is a general struct in case we want to add more vpn integrations
+type VPNInfo struct {
+	IPs       []net.IP
+	NodeID    string
+	VPNAPIKey string
```

Comment:
The VPN prefix on this stutters. Do you think all vpns will have a key, or should we have key+secret or user+password?
```suggestion
	APIKey string
```

Reply:
All VPNs have a JoinKey, which allows clients to register into the VPN mesh. APIKey is something required specifically for netmaker because the binary is not feature complete and you need to `curl` into the server for certain configs. You know what, let's remove that APIKey and if we ever integrate netmaker (or any other VPN that needs the workaround) we will add it